src/query/service/src/interpreters/interpreter_procedure_call.rs

```diff
@@ -0,0 +1,134 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use databend_common_ast::ast::DeclareItem;
+use databend_common_ast::ast::DeclareVar;
+use databend_common_ast::ast::Identifier;
+use databend_common_ast::ast::ScriptStatement;
+use databend_common_ast::parser::run_parser;
+use databend_common_ast::parser::script::script_block;
+use databend_common_ast::parser::tokenize_sql;
+use databend_common_ast::parser::ParseMode;
+use databend_common_exception::Result;
+use databend_common_expression::block_debug::box_render;
+use databend_common_expression::types::StringType;
+use databend_common_expression::DataBlock;
+use databend_common_expression::FromData;
+use databend_common_script::compile;
+use databend_common_script::Executor;
+use databend_common_script::ReturnValue;
+use databend_common_sql::plans::CallProcedurePlan;
+use databend_common_storages_fuse::TableContext;
+
+use crate::interpreters::util::ScriptClient;
+use crate::interpreters::Interpreter;
+use crate::pipelines::PipelineBuildResult;
+use crate::sessions::QueryContext;
+
+#[derive(Debug)]
+pub struct CallProcedureInterpreter {
+    ctx: Arc<QueryContext>,
+    plan: CallProcedurePlan,
+}
+
+impl CallProcedureInterpreter {
+    pub fn try_create(ctx: Arc<QueryContext>, plan: CallProcedurePlan) -> Result<Self> {
+        Ok(CallProcedureInterpreter { ctx, plan })
+    }
+}
+
+#[async_trait::async_trait]
+impl Interpreter for CallProcedureInterpreter {
+    fn name(&self) -> &str {
+        "ProcedureCall"
+    }
+
+    fn is_ddl(&self) -> bool {
+        false
+    }
+
+    #[fastrace::trace]
+    #[async_backtrace::framed]
+    async fn execute2(&self) -> Result<PipelineBuildResult> {
+        let res: Result<_> = try {
+            let mut src = vec![];
+            for (arg, arg_name) in self.plan.args.iter().zip(self.plan.arg_names.iter()) {
+                src.push(ScriptStatement::LetVar {
+                    declare: DeclareVar {
+                        span: None,
+                        name: Identifier::from_name(None, arg_name),
+                        default: arg.clone(),
+                    },
+                });
+            }
+            let settings = self.ctx.get_settings();
+            let sql_dialect = settings.get_sql_dialect()?;
+            let tokens = tokenize_sql(&self.plan.script)?;
+            let mut ast = run_parser(
+                &tokens,
+                sql_dialect,
+                ParseMode::Template,
+                false,
+                script_block,
+            )?;
+
+            for declare in ast.declares {
```

Comment:
May add tests for ScripBlock.declares

Reply:
You mean add test like this:
```sql
CREATE PROCEDURE p1(x UInt8, sum UInt8) RETURNS int not null LANGUAGE SQL COMMENT='test' AS $$
declare x := 20;
BEGIN
    FOR x IN x TO x + 3 DO
        sum := sum + x;
    END FOR;
    RETURN sum;
END;
$$;
```

I think this is a check and we need to do other check like this in next pr to enhance procedure.